### PR TITLE
Change obsreport receiver calling pattern in k8sclusterreciver, kubeletstatsreceiver, sapmreceiver

### PR DESCRIPTION
**Pull request by humivo**
**Description:** Change obsreport helpers for receiver to use the new pattern created in Collector. Specifically for k8sclusterreceiver, kubeletstatsreceiver, sapmreceiver.

**Link to tracking Issue:** [Issue #3436](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/3436)

### CODE REVIEW

File: receiver/sapmreceiver/trace_receiver.go [hunk 1]
```diff
@@ -208,10 +210,15 @@ func New(
 	if err != nil {
 		return nil, fmt.Errorf("failed to marshal default response body for %v receiver: %w", config.ID(), err)
 	}
+	transport := "http"
```
Comment (bogdandrutu) [inline receiver/sapmreceiver/trace_receiver.go:213]: This can be inlined in the New call

**Review by bogdandrutu — APPROVED** [review]
Please rebase

**Comment by bogdandrutu** [comment]
```
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jmxreceiver/subprocess	0.036s	coverage: 20.5% of statements
go tool cover -html=coverage.txt -o coverage.html
make[2]: Leaving directory '/home/circleci/project/receiver/jmxreceiver'
running make do-unit-tests-with-cover in ./receiver/k8sclusterreceiver
make[2]: Entering directory '/home/circleci/project/receiver/k8sclusterreceiver'
running go unit test ./... + coverage in /home/circleci/project/receiver/k8sclusterreceiver
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x228993c]

goroutine 324 [running]:
go.opentelemetry.io/collector/obsreport.(*Receiver).traceReceiveOp(0x0, 0x2cdb240, 0xc0004d81c0, 0x29e5099, 0x10, 0x0, 0x0, 0x0, 0xc0009006c0, 0xc000b00060)
	/home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.27.1-0.20210524201935-86ea0a131fb2/obsreport/obsreport_receiver.go:388 +0xdc
go.opentelemetry.io/collector/obsreport.(*Receiver).StartMetricsReceiveOp(...)
	/home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.27.1-0.20210524201935-86ea0a131fb2/obsreport/obsreport_receiver.go:301
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver.(*kubernetesReceiver).dispatchMetrics(0xc0001d4040, 0x2cdb240, 0xc0004d81c0)
	/home/circleci/project/receiver/k8sclusterreceiver/receiver.go:103 +0x1b9
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver.(*kubernetesReceiver).Start.func1(0xc0001d4040, 0x2cdb240, 0xc0004d81c0, 0x2cdb600, 0x3a38540)
	/home/circleci/project/receiver/k8sclusterreceiver/receiver.go:84 +0x377
created by github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver.(*kubernetesReceiver).Start
	/home/circleci/project/receiver/k8sclusterreceiver/receiver.go:58 +0x345
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver	1.228s
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/collection	0.064s	coverage: 85.6% of statements
?   	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/testutils	[no test files]
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/utils	0.042s	coverage: 50.0% of statements
FAIL
make[2]: *** [../../Makefile.Common:48: do-unit-tests-with-cover] Error 1
make[2]: Leaving directory '/home/circleci/project/receiver/k8sclusterreceiver'
```

File: receiver/sapmreceiver/trace_receiver.go (hunk 1 above)
Comment (humivo) [reply receiver/sapmreceiver/trace_receiver.go:213]: What do you mean by this can be inlined in the New call?

**Comment by humivo** [comment]
> ```
> ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/jmxreceiver/subprocess	0.036s	coverage: 20.5% of statements
> go tool cover -html=coverage.txt -o coverage.html
> make[2]: Leaving directory '/home/circleci/project/receiver/jmxreceiver'
> running make do-unit-tests-with-cover in ./receiver/k8sclusterreceiver
> make[2]: Entering directory '/home/circleci/project/receiver/k8sclusterreceiver'
> running go unit test ./... + coverage in /home/circleci/project/receiver/k8sclusterreceiver
> panic: runtime error: invalid memory address or nil pointer dereference
> [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x228993c]
> 
> goroutine 324 [running]:
> go.opentelemetry.io/collector/obsreport.(*Receiver).traceReceiveOp(0x0, 0x2cdb240, 0xc0004d81c0, 0x29e5099, 0x10, 0x0, 0x0, 0x0, 0xc0009006c0, 0xc000b00060)
> 	/home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.27.1-0.20210524201935-86ea0a131fb2/obsreport/obsreport_receiver.go:388 +0xdc
> go.opentelemetry.io/collector/obsreport.(*Receiver).StartMetricsReceiveOp(...)
> 	/home/circleci/go/pkg/mod/go.opentelemetry.io/collector@v0.27.1-0.20210524201935-86ea0a131fb2/obsreport/obsreport_receiver.go:301
> github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver.(*kubernetesReceiver).dispatchMetrics(0xc0001d4040, 0x2cdb240, 0xc0004d81c0)
> 	/home/circleci/project/receiver/k8sclusterreceiver/receiver.go:103 +0x1b9
> github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver.(*kubernetesReceiver).Start.func1(0xc0001d4040, 0x2cdb240, 0xc0004d81c0, 0x2cdb600, 0x3a38540)
> 	/home/circleci/project/receiver/k8sclusterreceiver/receiver.go:84 +0x377
> created by github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver.(*kubernetesReceiver).Start
> 	/home/circleci/project/receiver/k8sclusterreceiver/receiver.go:58 +0x345
> FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver	1.228s
> ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/collection	0.064s	coverage: 85.6% of statements
> ?   	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/testutils	[no test files]
> ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/k8sclusterreceiver/utils	0.042s	coverage: 50.0% of statements
> FAIL
> make[2]: *** [../../Makefile.Common:48: do-unit-tests-with-cover] Error 1
> make[2]: Leaving directory '/home/circleci/project/receiver/k8sclusterreceiver'
> ```

Looking into this right now!  I'm not sure why there is an issue though.

**Comment by bogdandrutu** [comment]
Please rebase :)

**Comment by humivo** [comment]
> Please rebase :)

I was advised to avoid rebasing by @Aneurysm9.  Is there any reason a rebase is needed?

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**